### PR TITLE
Document espnet tools and modules

### PR DESCRIPTION
**Pull request by ShigekiKarita**
This PR aims to generate doc on tools under `espnet/bin` and `utils`.
This PR is related to #876.

- [x] espnet/bin/*.py
- [x] utils/*.py
- [x] utils/*.sh
- [x] modules

online example (built by Travis CI in ShigekiKarita/espnet repo) https://karita.xyz/espnet/

### CODE REVIEW

**Comment by ShigekiKarita** [comment]
Many scripts are changed by a few lines but I believe there is no breaking change.

**Comment by ShigekiKarita** [comment]
After this PR is merged,  every PR has
- `utils/*.py` and `espnet/bin/*.py` without  `def get_parser() -> ArgumentParser`
- `utils/*.sh` without `--help` support

will be failed in Travis CI because I automated doc generation (at `ci/doc.sh`) by assuming these two points. Also I described this in `CONTRIBUTING.md`. This behaviour enforces us to write doc anytime.

**Comment by ShigekiKarita** [comment]
you can check how it looks like here (generated by `./ci/doc.sh`) http://karita.xyz/espnet-new-doc/apis/espnet_bin.html

File: utils/convert_fbank.sh [hunk 1]
```diff
@@ -14,20 +14,24 @@ win_length=
...
+  --cmd (utils/run.pl|utils/queue.pl <queue opts>) # how to run jobs.
+EOF
+)
 # End configuration section.
 
 echo "$0 $*"  # Print the command line for logging
 
 . parse_options.sh || exit 1;
 
 if [ $# -lt 1 ] || [ $# -gt 3 ]; then
```
Comment (kan-bayashi) [inline utils/convert_fbank.sh]:
```suggestion
if [ $1 == "--help" ] || [ $# -lt 1 ] || [ $# -gt 3 ]; then
```

**Comment by ShigekiKarita** [comment]
https://github.com/kaldi-asr/kaldi/blob/b96cab759caf411559d37d2c76b2af4008af61ed/egs/wsj/s5/utils/parse_options.sh#L24-L25

Yes. I updated `CONTRIBUTING.md` on `help_message="..."`.

**Comment by ShigekiKarita** [comment]
I am working on class autodoc in this PR because I expect it does not take hours...

**Comment by ShigekiKarita** [comment]
Package autodoc is DONE. You can check it here https://karita.xyz/espnet-new-doc/

**Comment by ShigekiKarita** [comment]
I completely removed warnings in `travis-sphinx build` caused by auto-generated files. Now, all the remaining warnings are docstring. Maybe these warnings can be removed in the later PRs.

**Comment by ShigekiKarita** [comment]
> What happens when we don't put def get_parser():?
results error when executing `./ci/doc.sh` (exit 1)

> v.0.4.1 or v.0.5.0? I personally think v.0.4.1 is better
agree. maybe need a few more works for v.0.5.0. I will create merge v0.4.1 -> v0.5.0 PR later.

**Comment by sw005320** [comment]
Also, can you change the following `1.0` to the current version?
(Is there a way to automatically do it by getting this from `setup.py`?)

image

**Comment by ShigekiKarita** [comment]
I have no idea at this time. We need more survey on how to connect doc and versions...

**Comment by ShigekiKarita** [comment]
Maybe we need to use a different service like https://docs.readthedocs.io/en/stable/index.html

**Comment by sw005320** [comment]
I thought that we can get the version info from
https://github.com/ShigekiKarita/espnet/blob/bin-doc/setup.py#L75
and put it in 
https://github.com/ShigekiKarita/espnet/blob/bin-doc/doc/conf.py#L80-L82

We could add such function in https://github.com/ShigekiKarita/espnet/blob/bin-doc/ci/doc.sh

**Comment by sw005320** [comment]
Can we also put a logo (https://github.com/ShigekiKarita/espnet/blob/bin-doc/doc/image/espnet_logo1.png) to the document?

**Comment by ShigekiKarita** [comment]
just a moment

**Comment by ShigekiKarita** [comment]
I'm still working on `utils/*.sh` doc problem in Travis CI...

**Comment by ShigekiKarita** [comment]
now it seems working! 
- log https://travis-ci.org/espnet/espnet/jobs/548190116#L1657
- generated page https://karita.xyz/espnet/apis/utils_sh.html

it is ready to merge

**Comment by sw005320** [comment]
Thanks a lot!
Now it's time to add more documents :)

**Comment by sw005320** [comment]
circleCI complains to me often for the gh-pages branch.
https://circleci.com/gh/espnet/espnet/2211?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification
Could you fix it?

**Comment by ShigekiKarita** [comment]
OK I will fix that.

**Pull request merged**